cumulusci/robotframework/Salesforce.py

```diff
@@ -485,32 +485,47 @@ def log_browser_capabilities(self, loglevel="INFO"):
         output += pformat(self.selenium.driver.capabilities, indent=4)
         self.builtin.log(output, level=loglevel)
 
-    def open_app_launcher(self):
+    @capture_screenshot_on_error
+    def open_app_launcher(self, retry=True):
         """Opens the Saleforce App Launcher Modal
 
         Note: starting with Spring '20 the app launcher button opens a
         menu rather than a modal. To maintain backwards compatibility,
         this keyword will continue to open the modal rather than the
         menu. If you need to interact with the app launcher menu, you
         will need to create a custom keyword.
+
+        If the retry parameter is true, the keyword will
+        close and then re-open the app launcher if it times out
+        while waiting for the dialog to open.
         """
-        locator = lex_locators["app_launcher"]["button"]
-        self.builtin.log("Clicking App Launcher button")
-        self._jsclick(locator)
 
-        api_version = int(float(self.get_latest_api_version()))
-        if api_version >= 48:
+        self._jsclick("sf:app_launcher.button")
+        self._jsclick("sf:app_launcher.view_all")
+        self.wait_until_modal_is_open()
+        try:
+            # the modal may be open, but not yet fully rendered
+            # wait until at least one link appears. We've seen that sometimes
+            # the dialog hangs prior to any links showing up
             self.selenium.wait_until_element_is_visible(
-                lex_locators["app_launcher"]["menu"],
-                error="Expected to see the app launcher menu, but didn't",
+                "xpath://ul[contains(@class, 'al-modal-list')]//li"
             )
-            element = self.selenium.get_webelement(
-                lex_locators["app_launcher"]["view_all"]
-            )
-            self.builtin.log("clicking 'view all' button")
-            self.selenium.capture_page_screenshot()
-            self._jsclick(element)
-        self.wait_until_modal_is_open()
+
+        except Exception as e:
+            # This should never happen, yet it does. Experience has
+            # shown that sometimes (at least in spring '20) the modal
+            # never renders. Refreshing the modal seems to fix it.
+            if retry:
+                self.builtin.log(
+                    f"caught exception {e} waiting for app launcher; retrying", "DEBUG"
+                )
+                self.selenium.press_keys("sf:modal.is_open", "ESCAPE")
+                self.wait_until_modal_is_closed()
+                self.open_app_launcher(retry=False)
+            else:
+                self.builtin.log(
+                    f"caught exception waiting for app launcher; not retrying", "DEBUG"
+                )
```

Comment:
Shouldn't we re-raise the exception here to prevent continuation of the test case, and to trigger the capture_screenshot_on_error mechanism?